system/lib/wasmfs/file.h

```diff
@@ -20,6 +20,43 @@
 
 namespace wasmfs {
 
+class path_string {
+  std::string path;
+
+public:
+  path_string() = default;
+  path_string(const std::string& p) : path(p) {}
+  path_string(const char* p) : path(p) {}
+  const std::string& str() const { return path; }
+};
+
+inline bool operator==(const path_string& x, const path_string& y) {
```

Comment:
I would implement these as methods rather than separate inline functions.

Reply:
I decided to make it non-member because it's a symmetrical operator.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c161-use-non-member-functions-for-symmetric-operators
But I can make them members. I think it doesn't matter since both operands have the same type.